---
title: "Lab 5 - Neutrality Statistics, Haplotype Homozygosity, and Selection"
author: "Christopher Schmitt"
date: "October 27, 2024"
output: 
  rmarkdown::html_document:
    theme: cosmo
---

```{r,echo=F}
library(knitr)
```

```{css,echo=F}
.sccCode {
background-color: black;
color: white;
}
```

## Introduction to Neutrality Statistics and Signs of Selection

***

### [Homework for Lab 5: DUE Friday, November 15th](https://fuzzyatelin.github.io/AN333_Fall24/Lab5_Homework.html)

***

**Readings**:<ul> 
<li>[Garrigan DR, Lewontin R, Wakeley J. 2010. Measuring the sensitivity of single-locus “neutrality tests” using a direct perturbation approach. *Mol Biol Evol* 27(1): 73-89.](https://academic.oup.com/mbe/article/27/1/73/1125718)</li><br>
<li>[Hughes DA, et al. 2009. Molecular evolution of UCP1 and the evolutionary history of mammalian non-shivering thermogenesis. *BMC Ecology & Evolution* 9: 4](https://bmcecolevol.biomedcentral.com/articles/10.1186/1471-2148-9-4)</li><br>
<li>[Yin N, Zhang D, Wang J. 2024. The *FTO* variant conferring enhanced *UCP1* expression is linked to human migration out of Africa. *Life Metabolism* 3(6): loae027](https://doi.org/10.1093/lifemeta/loae027)</li><br>
<li>[Sabeti PC, et al. 2002. Detecting recent positive selection in the human genome from haplotype structure. Nature 419(6909): 832-837.](https://www.nature.com/articles/nature01140)</li><br>
<li>[Tajima F. 1989. Statistical method for testing the neutral mutation hypothesis by DNA polymorphism. *Genetics* 123(3): 585–595.](http://www.genetics.org/content/123/3/585)</li></ul>

***

In this module, we we'll take our first look at site frequency spectrum *neutrality statistics*, which are statistical tests designed to measure if whether SNPs appear to be undergoing selection within a population. We'll go over the most popular of these statistics, **Tajima's D**, what it means, and how to interpret it, and then we'll use it to characterize our own data. We'll also take a look at *linkage-based* selection statistics, like *integrated haplotype scores* and *extended haplotype homozygosity*, which use similarly extended regions of LD across a population to infer the action of *selective sweeps* in the genome.
<br>
<br>
First, let's get an understanding of what each of these statistics might mean when calculated for our populations...

### Tajima's D
This is a fairly commonly used neutrality statistic. You can read the [original paper by Tajima](https://www.genetics.org/content/123/3/585), which details the development of his D statistic as a way of testing for the neutral mutation hypothesis (or Kimura’s theory of neutral mutation). *Tajima’s D* is a test statistic that compares the average number of pairwise differences in among base pairs in a population sample with the total number of variant sites in that sample. Most simply put, Tajima's D indicates whether the pattern of mutations found in a particular region of the genome in a population is following the assumptions of neutrality; in other words, if there has been selection for or against a particular allele in the population, the pattern seen should violate the assumptions of the neutral model.

Tajima's D is calculated using the relationship between the total number of variable loci in a given sequence in a population ($\theta$<sub>T</sub> which is related to S, which stands for the number of segregating sites) and the pairwise differences in that same sequence between each individual in the population ($\theta$<sub>T</sub>, or $\pi$).

When there are very few segregating sites, overall, along with very few pairwise differences between individuals, Tajima's D will be *positive*; such a lack of variation suggests that purifying selection is acting on this region of the genome. When there are more pairwise differences than segregating sites, Tajima's D is also positive, but in this case it indicates balancing selection.

An excess of low frequency polymorphisms will lead to a *negative* Tajima's D. This is a violation of neutrality suggesting that positive selection may have occurred, or perhaps a recent bottleneck or founder's events. In this case, we have multiple rare of an alleles against a background of very few segregating sites. This is often seen when one haplotype takes over a population and begins to accrue random mutations (as happens in a selective sweep or a founder/bottleneck event).

With Tajima’s D, the closer to 0 the D statistic is, the closer the locus being tested is to meeting the assumptions of neutrality.

Today, we'll calculate Tajima's D values for *our* populations in order to determine if any directional selection is happening in the *UCP1* region. 
<br>

### Integrated Haplotype Score (iHS)

Recent positive selection usually happens by *positive selective sweeps* (i.e. the rapid spread of a particular haplotype, driven by selection for particular alleles within that haplotype), which not only act upon the SNP that corresponds to the phenotype under selection, but the *entire haplotype* or *linkage region* in which that SNP is found. This can lead to *extended haplotype homozygosity (or EHH)* in a population, in which we see a high frequency of long, identical haplotypes in the genome around the SNP experiencing selection. An *iHS score*, or an *Integrated Haplotype Score*, is a test that can be used to measure recent positive selective sweeps around on particular allele indirectly by looking for evidence of extended haplotype homozygosity, while we can also directly measure extended haplotype homozygosity (EHH) itself in the population to assess hard selective sweeps. These tests identify *extended haplotypes* (large sets of alleles on a single region of the chromosome that are identical across individuals) and looking at how many of the haplotypes are homozygous within the population. Mass homozygosity across individuals for a particular haplotype is recorded as a *high* iHS score, while low levels of haplotype homozygosity is recorded as a *low* iHS score. We can (and will) also visualize EHH directly.<br> 

## Learning Outcomes

* Learn what Tajima's D, and iHS scores mean, and how to interpret them. 

* Learn to use the *pegas* package to calculate Tajima's D.

* Learn about iHS by calculate iHS and EHH in *R*, and reflect on what the iHS score for your population's *UCP1* region looks like. 

* Learn about whether or not (and what kind of) selection is happening in our populations based on these statistics, and relate it to environments that may cause these alleles in this region to be selected for or against.

## Part 1: Getting Started

Log in to the *SCC* and bring up the *R Studio* window [as usual](https://fuzzyatelin.github.io/AN333_Fall24/Lab2_Module.html).

Next, we need to install the package *PopGenome* in our *R Studio* space, because the SCC doesn't have it pre-installed.
```{r, eval=F, echo=T}
#We need to install this package
install.packages("rehh")
install.packages("pegas")
```

Next, load in the packages we need to use:
```{r, results='hide', message=F, warning=F}
library(vcfR)
library(rehh)
library(pegas)
```

Finally, we need to format our data for analysis. We'll make a *DNAbin* object ([as we have done before](https://fuzzyatelin.github.io/AN333_Fall24/Lab4_Module.html)) from our extended `UCP1_POP_REGION.vcf` file:
```{r}
YRI <- read.vcfR("UCP1_YRI_REGION.vcf")
YRIdna <- vcfR2DNAbin(YRI)
YRIdna
```

## Part 2: Tajima's D

Now we'll look at Tajima's D! We can get some useful information by using the `tajima.test` function in the {pegas} package. This function will use the *DNAbin* object that we created as an input. It will not only give us the D statistic, but also a *p-value* associated with the test. This is important, because it will give us a sense of how significant the results are of our neutrality test. 
<br>
```{r}
tajima <- tajima.test(YRIdna)
tajima
```

What we see from YRI for Tajima's D is that this population does not appear to be subject to selection in the extended *UCP1* gene region. This suggests that if there is *evolution* happening in the YRI population's *UCP1* region (remember, that's a change in allele frequencies over generations, and does *not* have to be due to selection), then it is likely selectively neutral. 
<br>
<br>
Interpret your own population's Tajima's D value. What is the D statistic for your population, and what does it mean in terms of whether and what kind of selection might be occurring on your population? We will revisit this question at the end of the module.

## Part 3: Positive selective sweeps using iHS and EHH

Now, we will run our *iHS* and *EHH* analyses to assess recent positive selective sweeps in an expanded area surrounding *UCP1* on chromosome 4 (NOTE: you MUST download a new VCF file to run this analysis). To run this example, I'm using the *rehh* package, which is specifically designed to look for *Extended Haplotype Homozygosity*, or evidence of hard positive selective sweeps.

To do this, we'll first need to load our *modules*. This will include *tabix* to download the data, *vcftools* and *bcftools* to modify it:

```{bash, eval = F, echo = T, class.source="sccCode"}
module load htslib/1.16
module load vcftools
module load bcftools/1.16
```

Now, to save on space, we'll download our extended positions (roughly 1 Mb above and below our gene region of interest) from chromosome 4 into a *zipped* file using a familiar *tabix* command. You might note that the web FTP where we're downloading this from is different from in Module 01... this is because to estimate EHH we need an *ancestral allele* estimate; these were included in earlier versions of the 1KG data, but not for the latest iteration:
```{bash, eval = F, echo = F, class.source="sccCode"}
# I feel like I'm going insane... I got AA from the URL in the next chunk one night, and the next morning the VCF did NOT have the AA INFO tag... but in this chunk this URL has the tag but is mapped to HG37... so the positions don't match the rest of the course. Keeping it here for reference.
tabix -h https://ftp-trace.ncbi.nih.gov/1000genomes/ftp/release/20130502/ALL.chr4.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz  4:139559434-141568805 | bgzip -c > UCP1_YRI_EXTENDED_all.vcf.gz
```
```{bash, eval = F, echo = T, class.source="sccCode"}
tabix -h https://ftp-trace.ncbi.nih.gov/1000genomes/ftp/release/20130502/supporting/GRCh38_positions/ALL.chr4.phase3_shapeit2_mvncall_integrated_v3plus_nounphased.rsID.genotypes.GRCh38_dbSNP_no_SVs.vcf.gz 4:139559434-141568805 | bgzip -c > UCP1_YRI_EXTENDED_all.vcf.gz
```

Then, we'll subset the zipped file for *just* the YRI data with a familiar *vcftools* command. While we're at it, we'll also filter out all indels and SNPs with a MAF lower than 5% while keeping the ancestral allele information:
```{bash, eval = F, echo = T, class.source="sccCode"}
vcftools --gzvcf UCP1_YRI_EXTENDED_all.vcf.gz --keep YRI.samples.list --maf 0.05 --remove-indels --recode-INFO AA --recode --stdout | bgzip -c > UCP1_YRI_EXTENDED.vcf.gz
```

You can take a look to make sure this went well (i.e., you've got *your* sample individuals and the right subset of loci) using the following code (press `q` to leave the `less` view when you're done):
```{bash, eval = F, echo = T, class.source="sccCode"}
gunzip -c UCP1_YRI_EXTENDED.vcf.gz | less
```

Ok... now let's open *Rstudio* and try it!

Once you're in *R*, load the {rehh} library (if it throws an error, you may need to install {rehh}, which you can do through the `Packages` tab):
```{r}
library(rehh)
```

Then, we need to import our newly made `Extended` VCF file for our population:
```{r,eval=F, echo=F}
#This only works with the INFO AA tags... see the last semester's processing information to get these files.
hap<-data2haplohh(hap_file="hapUCP1_YRI.out",map_file="mapUCP1_YRI.out",chr.name="4")
```
```{r}
hap<-data2haplohh(hap_file="UCP1_YRI_EXTENDED.vcf.gz",polarize_vcf = FALSE)
```

Once those are read in, you should see that our dataset consists of 6052 markers, consisting of 6046 SNPs split into 216 haplotypes.

Now we can run our extended haplotype homozygosity assessment (this step might take a while, be patient):
```{r}
ehh<-scan_hh(hap,limhaplo=2,limehh=0.05,limehhs=0.05,maxgap=NA,threads=1)
```

Retrieve our iHS scores:
```{r}
ihs <- ihh2ihs(ehh,freqbin=0.025)
```

We can then look for candidate regions potentially under selection using the iHS score. Here we’re considering a piHS (on the y-axis) greater than 2 as a significant score:
```{r}
cr.se<-calc_candidate_regions(ihs,threshold=2,pval=TRUE,window_size=3000,overlap=300,min_n_extr_mrk=1)
cr.se
```
Looks like, in the YRI population, we have perhaps 31 regions that have potentially experienced a hard selective sweep.

Let's plot them:
```{r}
manhattanplot(ihs,pval=TRUE,threshold=2,pch=16, chr.name="4",cr=cr.se,cr.col="skyblue",main="iHS")
```

As you can see, there are a lot of potential positive selective sweeps across this extended region!  However, we also need to remember... in this figure we've added 1 Mb (a million bases) above and below our *UCP1* region, so the only spots of selection that are relevant are between 4:140559434-140568805 (the *UCP1* region) and 10 kb up and downstream of them... let's look at those, specifically:
```{r}
manhattanplot(ihs, pval=TRUE, threshold=2, pch=16, chr.name="4", cr=cr.se, cr.col="skyblue", main="iHS", xlim=c(140549434,140578805))
```

As you can see, in the YRI population it looks like there's evidence for one clear positive selective sweep in the upstream area of *UCP1* region.

We can also try to map our SNPs of interest from Module 3 to the figure. We'll start by making a dataframe with the position information for these loci of interest, called `loi`, which gives thier VariantID, chromosome, and position:
```{r}
CHR<-c(4,4,4,4,4)
POSITION<-c(140560427,140562317,140563980,140569265,140572807)
loi<-as.data.frame(cbind(CHR,POSITION))
rownames(loi)<-c("rs6536991","rs2270565","rs12502572","rs3811787","rs1800592")
loi
```

Then, we can insert this information into our figure using the series of `mrk` options, below:
```{r}
manhattanplot(ihs, pval=TRUE, threshold=2, pch=16, chr.name="4", cr=cr.se, cr.col="skyblue", main="iHS", xlim=c(140549434,140578805), mrk=loi, mrk.col="blue", mrk.pch=20, mrk.lab.pos=3)
```

You may notice that one of our SNPs of interest, *rs2270565*, is missing here (and, indeed, that it looks like we don't have all 117 of the SNPs we should). This may be because some of these had a low minor allele frequency (MAF < 0.05). You may also notice that our SNPs of interest that are present appear to have relatively low iHS scores... suggesting that, if they are under selection, it's likely not a hard selective sweep.

Let's actually check the SNPs making up these significant regions:
```{r}
library(tidyverse)
pihs<-as_tibble(ihs$ihs[,c(2,4)])

sig.pihs<-
  pihs %>%
  filter(LOGPVALUE >= 2) %>%
  arrange(LOGPVALUE)
sig.pihs
```

Now we can take a closer look at the EHH of an actual *SNP* of interest. Let's try some of our SNPs of interest from [Module 3](https://fuzzyatelin.github.io/AN333_Fall24/Lab3_Module.html): *rs6536991* (4:140560427), *rs2270565* (4:140562317), *rs12502572* (4:140563980), *rs3811787* (4:140569265), and *rs1800592* (4:140572807).

Let's start with the famous -3826A<G SNP, *rs1800592*.

First, though, let's get a sense of what we're looking for... we can look at some of these significant SNPs within the region of interest to set expectations. Let's take a look at 140343413, also known as [*rs9996243*](https://www.ncbi.nlm.nih.gov/snp/?term=4%3A140343413), which has among the higher p<sub>iHS</sub> in our extended region (p<sub>iHS</sub> = 4.073600). In this case, I'm zooming in on 60 kb up- and down-stream of the focal SNP using the `xlim` option: 
```{r}
ehh_140343413<-calc_ehh(hap,mrk="rs9996243",polarized=FALSE,include_nhaplo=TRUE)
plot(ehh_140343413,
     main="EHH at 4:140343413",
     col=c("blue2","gold2"))
```

Wow... we can see here that EHH (on the y-axis) for the haplotypes surrounding the minor allele is consistently higher than the EHH surrounding the major allele, with that EHH being consistently high as we approach the SNP along the genome, particularly upstream of our focal SNP. This suggests that recombination has not yet broken up the linkage block/haplotype surrounding the ancestral allele anywhere where the EHH is equal to 1. In other words, it would appear that the minor allele for our SNP here is in a *very* long haplotype with extensive haplotype homozygosity, suggesting that the minor allelle associated SNPs have largely been shielded from recombination.

Let's actually visualize the haplotype structure around this SNP of interest to give us a better idea of the observed effect of selection. The original use of this kind of graph - called a *bifurcation graph* - is in a paper by [Sabeti et al. (2002)](https://www.nature.com/articles/nature01140), in our readings for this module. This will look like a bifurcating tree, with the thicker branches representing the relative number of individuals maintaining that particular haplotype (thicker = more people in the sample have that haplotype) surrounding the core allele (which is at the vertical dashed line). The nodes of the tree are where the haplotype/linkage is broken by a variant outside the original haplotype. Let's check it out:
```{r}
furcation_140343413<-calc_furcation(hap,mrk="rs9996243",polarized=FALSE)
plot(furcation_140343413,
     lwd=0.1,
     col=c("blue2","gold2"),
     cex.lab=0.3,
     main="Bifurcation 4:140343413",
     legend.xy.coords="none")
```

Here, we can see that there are a lot of individuals in the YRI population that have the major allele, but the haplotype breaks down pretty quickly (or the haplotypes *bifurcate* each time anyone has a mutation as you move away from the focal SNP) but that there's some clear extensive haplotype homozygosity around the minor allele (no bifurcations means no mutations). We can see this because the thicker the branch, the more individuals share haplotypes along that branch. So when we see a very thick branch extend many Mb along the genome, this suggests a hard selective sweep. In this case, it perhaps looks like a relatively limited selective sweep, given that not eveyrone has this haplotype. For a very clear selective sweep, we would see almost the entire population showing the extended haplotype (e.g., a very large, thick line extending across a large portion of the genomic region).

We can also zoom in on the EHH block using the `xlim` option:
```{r}
ehh_140343413<-calc_ehh(hap,mrk="rs9996243",polarized=FALSE,include_nhaplo=TRUE)
plot(ehh_140343413,
     main="EHH at 4:140343413",
     col=c("blue2","gold2"),
     xlim=c(140323413,140583413))
```
```{r}
furcation_140343413<-calc_furcation(hap,mrk="rs9996243",polarized=FALSE)
plot(furcation_140343413,
     lwd=0.1,
     col=c("blue2","gold2"),
     cex.lab=0.3,
     main="Bifurcation 4:140343413",
     legend.xy.coords="none",
     xlim=c(140323413,140583413))
```

Now, let's look at a random SNP we for which we have no reason to believe there is *no* selective sweep. The position 4:140560607, also known as [*rs726989*](https://www.ncbi.nlm.nih.gov/snp/?term=4%3A140560607), in the *UCP1* region has a very low p<sub>iHS</sub> value (p<sub>iHS</sub> = 0.0391). Let's take a look. Here, I'll zoom in to ~30 kb up- and down-stream of this SNP using `xlim`:
```{r}
ehh_140560607<-calc_ehh(hap,mrk="rs726989",polarized=FALSE,include_nhaplo=TRUE)
plot(ehh_140560607,
     main="EHH at 4:140560607",
     col=c("blue2","gold2"),
     xlim=c(140530607,140590607))
```

Ok, this looks like another case where the minor allele appears to have slightly more haplotype homozygosity than the major allele (e.g., the gold line is higher than the blue line across most of the region), but that EHH doesn't extend very far from our SNP of interest before falling away, suggesting that there is no selection keeping recombination from breaking up the haplotypes around our focal SNP.

Let's look at the bifurcation diagram:
```{r}
furcation_140560607<-calc_furcation(hap,mrk="rs726989",polarized=FALSE)
plot(furcation_140560607,
     lwd=0.1,
     col=c("blue2","gold2"),
     cex.lab=0.3,
     main="Bifurcation 4:140560607",
     legend.xy.coords="none",
     xlim=c(140530607,140590607))
```

As you can see by the extensive branching, our haplotypes around both the ancestral and the derived allele are breaking down pretty quickly at this locus! There does appear to be a bit of haplotype homozygosity around the ancestral allele, but it's not so extensive compared to the derived (nor compared to our top hit, above!).

Now, let's try our SNP of interest, -3826A<G or *rs1800592* (4:140572807).	

```{r}
ehh_140572807<-calc_ehh(hap,mrk="rs1800592",polarized=FALSE,include_nhaplo=TRUE)
plot(ehh_140572807,
     main="EHH at 4:140572807",
     col=c("blue2","gold2"),
     xlim=c(140552807,140592807))
```

```{r}
furcation_140572807<-calc_furcation(hap,mrk="rs1800592",polarized=FALSE)
plot(furcation_140572807,
     lwd=0.1,
     col=c("blue2","gold2"),
     cex.lab=0.3,
     main="Bifurcation 4:140572807",
     legend.xy.coords="none",
     xlim=c(140552807,140592807))
```

How would you interpret these graphs?  Is there EHH?  How extensive is it?  Can you find the p<sub>iHS</sub> for this SNP?  How about for the other SNPs of interest from Module 3? 

Try this with *your* population. Are there any SNPs that look like clear positive selective sweeps in the *UCP1* region? Can you infer from the SNP positions and *EnsEMBL* if these are in any functional regions?

## Part 5: What Do Your Results Mean? Discuss with a partner from class:

As we wrap up this lab, think about the results you've generated. You have already been prompted to think about what your results might mean as we've gone along; now you have the opportunity to talk to a partner about what you think your results might mean.

* What is the Tajima's D statistic and accompanying P-value for your population, and what does that mean regarding potential selection happening on your population?

* What do you think of the iHS scores in *UCP1* in your data? Does this fit with your results based on Tajima's D? Based on what we've discovered today about selection happening in our populations, do you think that a recent positive selective sweep may have happened in *UCP1* for you population? Is there anything that you know about the climate and history of your population that you can use to aid in understanding why there may or may not have been a recent positive selective sweep in your population in this region?
<br>
<br>
<br>
<br>
<br>
<br>

# Extra Credit: Sliding window Tajima's D!

A perhaps more helpful method for assessing Tajima’s D can be done using {vcftools} in the SCC/Terminal workspace. This version analyses Tajima’s D in small sliding windows (here, I've chosen 500 bp windows) along the gene region of interest, allowing us to better pinpoint exactly where in the gene region selection may be occurring:
```{bash, eval = F, echo = T, class.source="sccCode"}
module load vcftools
vcftools --gzvcf UCP1_YRI_EXTENDED.vcf.gz --TajimaD 500 # We chose 500 as the window size
mv out.Tajima.D UCP1_YRI_REGION.Tajima.D #rename the output file 
mv out.log UCP1_YRI_REGION.TD.log #rename log file 
less UCP1_YRI_REGION.Tajima.D 
```

With this sliding window, we can see that certain sections of our severe COVID-19 risk region have very different values for Tajima’s D. A good rule of thumb is that a Tajima's D value above 2 or below -2 is significant. For example, the 500 bp windows starting at position 45823000 has 4 SNPs in it and is negative but not quite significant (D = -1.56823000).

Let’s take a closer look by bringing the output from this analysis into *R* so that we can filter it according to this significance threshold:
```{r}
slide.ucp1<-read.table("UCP1_YRI_REGION.Tajima.D",header=TRUE)
head(slide.ucp1)
```

Now we can filter it by our significance threshold using the {tidyverse} package:
```{r,warning=F,message=F}
library(tidyverse)
slide.ucp1.sig<-
  slide.ucp1 %>%
  filter(TajimaD <= -2 | TajimaD >= 2) %>%
  arrange(TajimaD)
slide.ucp1.sig
```

Here, we can see that we have 232 (!) 500-bp windows in the extended *UCP1* region that have significant Tajima's D values for the Yoruban population. All of these values are positive, suggesting purifying selection on at least one of the SNPs in each of these regions. The most significant region is **4:140922000-140922500** (the last row in our table), and there are 6 SNPs in that region that could be the potential focus for selection. Going back to *EnsEMBL*, it looks like this 500 bp window is in an *intronic* region of the gene *RNF150*, so if it is experiencing purifying selection, it is most likely related to the function of this particular gene (although, of course, it may also simply be in high LD with a functional region elsewhere). I found this functional information by searching for the 500 bp window, 4:140922000-140922500, in the human reference genome on *EnsEMBL* and looking at the **Region In Detail** and **Variant Table**. Remember, if your population had no variants at certain SNPs, then those SNPs were dropped from your VCF file; this means there may be more SNPs listed on *EnsEMBL* than are actually in your sliding window for your population.

Let's check out *rs184151696* (4:140922224), which is an intronic variant in our window (4:140922000-140922500) that shows the highest value for Tajima's D:
```{r}
ehh_140922224<-calc_ehh(hap,mrk="rs184151696",polarized=FALSE,include_nhaplo=TRUE)
plot(ehh_140922224,
     main="EHH at 4:140922224",
     col=c("blue2","gold2"),
     xlim=c(140894224,140950224))
```

Definitely looks like there is *no* differential extensive haplotype homozygosity between the ancestral and derived alleles. Let's look at the bifurcation diagram:
```{r}
furcation_140922224<-calc_furcation(hap,mrk="rs184151696",polarized=FALSE)
plot(furcation_140922224,
     lwd=0.1,
     col=c("blue2","gold2"),
     cex.lab=0.3,
     main="Bifurcation Diagram\nrs184151696 in YRI",
     legend.xy.coords="none",
     xlim=c(140894224,140950224))
```

Hmmm... looks like a lot of bifurcation, suggesting a lot of broken haplotypes (with a little EHH around the major allele).

What regions show significantly positive or negative Tajima's D in your population?  Are they the same as in the YRI population? Are they implicated in similar traits or genomic functions?

Why do you think there can be significant site frequency spectrum-based evidence of selection in a genomic region, but NOT linkage-based evidence of a positive selective sweep?  How are these two processes different, and how do they lead to different patterns in a genomic region around the variant specifically under selection?

<br>
<br>
<br>
<br>
<br>
<br>