---
title: "Midterm 2"
author: "Christopher A Schmitt"
date: "November 7, 2024"
output: 
  rmarkdown::html_document:
    theme: cosmo
---

## Midterm 2 (150 points)

This midterm is **open notes, open textbook, open Lab tutorial** and will cover genetics review concepts (insofar as they are relevant to our newer work), linkage disequilibrium, genetic drift, heterozygosity, and detecting and measuring selection in the genome. <br>

You may begin the exam when the submission link (below) becomes active at class time on **Monday, November 8**, and work on the exam until the link closes at midnight of the exam due date, **Friday, November 15**.<br>

*No late responses will be accepted!*<br><br>

<center>
### [Access the online interface to submit your Midterm 2 answers here.](https://forms.gle/xuoRmLV8ubGjukPdA)
</center><br>

Parts of this exam will require downloading, processing, and analyzing data from *EnsEMBL* in ways that should, by now, be familiar to you; these may involve interfacing with *R/R Studio*, *tabix*, *vcftools*, and other modules in the *SCC*. If you've forgotten how to use these modules, you should use your online module tutorials as guides (this is an open notes exam, after all).<br>

All *SCC*-based files associated with exam analyses should be uploaded to your named a folder in the general `/projectnb/anth333/` project space. Doing this will be part of the exam, and involves commands you've already learned and used in the context of class.<br>

NOTE: I will *not* help you directly with exam questions in office hours or during our class time with the midterm. However, I *will* help you with exercises already present on Lab tutorials or previous homework assignments. *Please do not ask for help on exam questions directly*. Also, although you are encouraged to do your *homework* in cooperation with other students, you should be doing your take-home exam *alone*.<br>

***

###Part 1: Preparing your Workspace for Midterm 2 (5 points)

<ol>1. Log in to your *SCC* working directory in the `/projectnb/anth333/` project space. **(1 point)**<br><br>

<ol>2. In the directory `/projectnb/anth333/`, you should already have a shared project space named after your BU login. In that directory, create a new directory named '**Midterm2**'. **(2 points)**</ol><br>

<ol>3. Now, navigate to your newly named directory so that it's your current directory (i.e., where all of the files you process in *R* will be deposited and saved). *Conduct all work for the midterm from this directory*. At the end of the exam, all your newly created or saved files associated with Midterm 2 *must* be in this folder. **(2 points)**<br><br>

HINT: Given you answered these first questions correctly, your *SCC* prompt will look like this for all midterm analyses:

```{bash, eval = F, echo = T}
[username@scc1 Midterm2]$
```

And your *SCC On Demand* file pane and console pathways should look like this:
```{r,eval=F}
~/projectnb/anth333/username/Midterm2/
```

Now, all of the analyses and processing you do will be done *within* this single named directory, allowing me to grade your individual progress.</ol>

***

###Part 2: Linkage in a New Population (40 points)

So far in our labs, we've been focusing on the human *UCP1* gene in our own assigned *1000 Genomes Project* populations. To better demonstrate what we've learned, let's take a look at the *UCP1* gene region in a population where we know cold to be a potentially strong selective force: the *Finnish from Finland*.<br><br>

Now, I'm already convinced that we all know how to properly download a dataset from *EnsEMBL*, so please just copy your VCF file for the **Finnish in Finland** from the `sampleVCF` folder. Make sure it's the version that includes 10,000 bp up- and down-stream of the gene region (i.e., the file called `UCP1_FIN_REGION.vcf`. **(5 points)**<br><br>

HINT: If you're having trouble with the `cp` command (last seen in [Module 4](https://fuzzyatelin.github.io/AN333_Fall24/Lab4_Module.html)), you can always go back to [Module 1](https://fuzzyatelin.github.io/AN333_Fall24/Lab1_Module.html) to create this region from scatch!

The next few questions will have to do with the concept of *linkage*.

<ol>1. Which process in the genome is responsible for driving linkage between any two loci towards equilibrium, and when does it happen? **(5 points)**

<ol>a) Mutation, during fertilization</ol>

<ol>b) Drift, during mitosis</ol>

<ol>c) Recombination, during meiosis</ol>

<ol>d) Selection, during evolution</ol>
</ol><br>

<ol>2. Imagine you conduct a dihybrid cross between pea plants, which can have either purple or red flowers (with the red flowers being recessive). Which of the following results in your F2 generation of this dihybrid cross indicate linkage (for the given sample size)? **(2 points)**

<ol>a) 281 : 94 : 94 : 31 (n = 500)</ol>

<ol>b) 197 : 66 : 65 : 22 (n = 350)</ol>

<ol>c) 215 : 71 : 71 : 24 (n = 381)</ol>

<ol>d) 327 : 27 : 27 : 55 (n = 436)</ol>
</ol><br>

<ol>3. Although it is generally recognized that *mitochondrial DNA* does *not* recombine, what evidence seems to suggests that might not be the case? **(2 points)**

<ol>a) Recombination in a large sample of human mtDNA seems to vary as a function of the number of bases between two loci.</ol>

<ol>b) There appears to be recombination between mtDNA haplotypes in humans with heteroplasmy.</ol>

<ol>c) Heteroplasmic mtDNA lineages can be passed on from parent to offspring.</ol>

<ol>d) All of the above.</ol>
</ol><br>

<ol>4. Which of the following loci on the Y chromosome are linked? **(2 points)**

<ol>a) *CDY2A* and *DAZ3*</ol>

<ol>b) *USP9Y* and *BPY2C*</ol>

<ol>c) *TSPY4* and *TSPY8*</ol>

<ol>d) All of the above; all loci on the Y chromosome (outside the PAR) are linked because it doesn't recombine.</ol>
</ol><br>

<ol>5. What is the measure of genetic distance on a *linkage map*? **(2 points)**

<ol>a) Megabases (Mb)</ol>

<ol>b) CentiMorgans (cM)</ol>

<ol>c) Nei's Distance (N)</ol>

<ol>d) None of the above.</ol>

</ol><br>

<ol>6. When comparing recombinant to non-recombinant offspring in a population of interest, you calculate a LOD score of 5.2 for two particular loci. In your population, are these two loci in linkage disequilibrim? **(2 points)**

<ol>a) Yes.</ol>

<ol>b) No.</ol>

<ol>c) It's impossible to say from this information.</ol>

</ol><br>

<ol>7. Imagine you measure the gametic genotypes for two monogenic traits (seen in the parental haplotype as AB/ab) in proportions 0.48, 0.48, 0.02, 0.02 (AB, ab, Ab, aB). Are these two loci experiencing linkage disequilibrium? **(2 points)**

<ol>a) Yes.</ol>

<ol>b) No.</ol>

<ol>c) It's impossible to say from this information.</ol>

</ol><br>

<ol>8. What is *D*, the measure of LD, for the problem above? **(2 points)**</ol><br>

For the next few questions, we'll interrogate the dataset we copied for the extended *UCP1* gene region the *Finnish in Finland* population from the *1000 Genomes Project* dataset. Using the methods in [Module 3](https://fuzzyatelin.github.io/AN333_Fall24/Lab3_Module.html), create an LD heatmap for the Finnish in Finland using the **r<sup>2</sup>** test statistic. <br>

<ol>9. Upload your *r<sup>2</sup>* heatmap for the entended *UCP1* gene region for the *Finnish in Finland* population **(10 points)**</ol><br>

<ol>10. Which of the *UCP1* SNPs of interest from the LD module - *rs6536991* (4:140560427), *rs2270565* (4:140562317), *rs12502572* (4:140563980), *rs3811787* (4:140569265), and *rs1800592* (4:140572807) - are in linkage disequilibrium (LD > 0.8) in the *Finnish in Finland* population? **(6 points)**</ol><br>

###Part 3: Genetic Drift (10 points)

<ol>11. Imagine you're studying a small population of critically endangered yellow-tailed woolly monkeys (*Lagothrix flavicauda*) in Per√∫. You manage to genotype the 8 remaining individuals in an isolated population in the Andes at SNP rs16534 (C/A), and find it's in Hardy-Weinberg equilibrium with the frequency of the C allele at 0.35. A few years later, you return and genotype the *next generation* of 8 individuals. Given this information, what is the probability that this next generation of yellow-tailed woolly monkeys will be fixed for the C allele at rs16534? **(4 points)**</ol><br>

<ol>12. Which of the following is a potential consequence of this reduced population size in yellow-tailed woolly monkeys? **(2 points)**

<ol>a) Reduced total number of alleles.</ol>

<ol>b) Changes in allele frequency.</ol>

<ol>c) Increased linkage disequilibrium.</ol>

<ol>d) None of the above.</ol>

</ol><br>

<ol>13. If the remnant yellow-tailed woolly monkey population you studied had only 2 males and 6 females, what is the *effective population size (N<sub>e</sub>)*? **(2 points)**</ol><br>

<ol>14. In the second generation of the remnant yellow-tailed woolly monkey population you studied, you genotype them to better understand how many offspring that generation were from each of the six females. You find that there's a rather large variance in the number. Would that increase or decrease the effective population size? **(2 points)**</ol><br>

###Part 4: Mutations, Heterozygosity, and Selection (80 points)

<ol>15. In your yellow-tailed woolly monkey population, given that the SNP rs16534 is fixed for the C allele, and a mutation rate of 0.00025, what would the frequency of the C allele be after 10,000 generations? **(2 points)**</ol><br>

<ol>16. In which mutation model, in which each mutation is considered to be novel, does one simply count the number of difference and assume that represents the total number of mutations that have occured? **(2 points)**

<ol>a) Infinite alleles model</ol>

<ol>b) Stepwise mutation model</ol>

<ol>c) Nei's genetic distance</ol>

<ol>d) None of the above.</ol>

</ol><br>

<ol>17. Which mutation model is the preferred model for microsatellite loci? **(2 points)**

<ol>a) Infinite alleles model</ol>

<ol>b) Stepwise mutation model</ol>

<ol>c) Nei's genetic distance</ol>

<ol>d) None of the above.</ol>

</ol><br>

<ol>18. In Module 4, we used the Neighbor-Joining method to construct a tree based on an infinite alleles mutation model. Does the neighbor-joining method always give you the evolutionarily 'correct' tree? Why or why not? **(5 points)**</ol><br>

<ol>19. Please take a screenshot or save a PNG or PDF file of a neighbor-joining tree for the *Finnish in Finland* population (as done in Module 4) and upload it here. Your file size must not exceed 10 MB. **(10 points)**</ol><br>

<ol>20. If we align several individual sequences, and see a site that has multiple bases at that locus, what do we call it (of the following options)? **(2 points)**

<ol>a) Segregating site</ol>

<ol>b) Non-conserved site</ol>

<ol>c) Polymorphic site</ol>

<ol>d) All of the above</ol>

</ol><br>

Imagine that we all, as a class, submitted samples to the Sensory Morphology and Anthropological Genomics Lab for sequencing of our individual *UCP1* genes. For a 20-bp sequence, we get the following information regarding *segregating sites* in that region:<br><br> 

<center>
![](img/hetsel.png)
</center><br>

<ol>21. What is the *observed heterozygosity* (—≤<sub>œÄ</sub>) of this sample?  **(2 points)**</ol><br>

<ol>22. The *estimated heterozygosity* (—≤<sub>k</sub>) of this sample is 0.86. Given this, what would we calculate Tajima's D to be for this region of the genome in our class population? **(2 points)**

<ol>a) Positive</ol>

<ol>b) Zero</ol>

<ol>c) Negative</ol>

</ol><br>

<ol>23. Given the value of Tajima's D for our class population, what kind of selection might be occurring in our classroom!? **(2 points)**

<ol>a) No selection</ol>

<ol>b) Diversifying selection</ol>

<ol>c) Purifying selection</ol>

<ol>d) Balancing selection</ol>

</ol><br>

<ol>24. We also calculated Tajima's D, in a different and much faster way, in [Module 5](https://fuzzyatelin.github.io/AN333_Fall24/Lab5_Module.html). Calculate Tajima's D for the *Finnish in Finland* population for our extended *UCP1* region. **(5 points)**</ol><br>

<ol>25. Is this value for Tajima's D statistically significant? **(2 points)**

<ol>a) Yes</ol>

<ol>b) No</ol>

</ol><br>

<ol>26. Given the value of Tajima's D for the *Finnish in Finland*, what kind of selection might be occurring around *UCP1* in this population? **(2 points)**

<ol>a) No selection</ol>

<ol>b) Diversifying selection</ol>

<ol>c) Purifying selection</ol>

</ol><br>

When working in South Africa, you collect genomic data on three populations of vervet monkey (Pop1, Pop2, and Pop3). Each individual is genotyped at a particular SNP, the genotypes for which are summarized in the following table:<br><br>

<center>
![](img/fst.png)
</center><br>

Answer the following questions based on these genotype results.

<ol>27. What is the observed heterozygosity (H<sub>obs</sub>) and expected heterozygosity (H<sub>exp</sub>) of each population? **(6 points)**</ol><br>

<ol>28. Which population has a positive F-value? **(2 points)**</ol><br>

<ol>29. What does a positive F-value imply for that population? **(2 points)**</ol><br>

<ol>30. According to the F<sub>ST</sub> for these populations, what is the proportion of variation at this locus that is due to differences between subpopulations? **(2 points)**</ol><br>

<ol>31. What kind of sequencing is best for identifying and thus reducing errors in variant/polymorphism discovery with whole-genome sequencing? **(2 points)**

<ol>a) Sanger sequencing</ol>

<ol>b) Bisulfite sequencing</ol>

<ol>c) Duplex sequencing</ol>

<ol>d) Single-strand sequencing</ol>

</ol><br>

<ol>32. The *EDAR* gene is a gene implicated in numerous phenotypes including hair thickness, sweat gland output, tooth morphology, and mammary duct proliferation. What is this phenomenon called, where one gene is implicated in multiple seemingly unrelated phenotypes? **(2 points)**

<ol>a) Epistasis</ol>

<ol>b) Pleiotropy</ol>

<ol>c) Epigenetics</ol>

<ol>d) Polygenics</ol>

</ol><br>

<ol>33. The McDonald-Kreitman Test is a test designed to detect taxon-informative markers; we use it to contrast *within* and *between* taxon synonymous and non-synonymous differences. In the case of *Drosophila* variation at the alcohol dehydrogenase gene (see the figure on slide 23 of Lecture 12), which of the loci presented are the *best* species-informative loci to differentiate between them (e.g., loci that really differentiate species from each other) aside from 781, which we discussed in class? **(6 points)**</ol><br>

<ol>34. When using d<sub>N</sub>/d<sub>S</sub>, a significantly *positive* value implies what kind of selection? **(2 points)**

<ol>a) Positive</ol>

<ol>b) Negative</ol>

<ol>c) Neutral</ol>

<ol>d) Purifying</ol>

</ol><br>

<ol>35. Which of the following physiological systems appear to have been under significant positive selection since our last common ancestor with chimpanzees, according to d<sub>N</sub>/d<sub>S</sub>? **(2 points)**

<ol>a) Immunity and defense</ol>

<ol>b) Digestion</ol>

<ol>c) Olfaction</ol>

<ol>d) Brain growth</ol>

</ol><br>

<ol>36. Are hard selective sweeps common in human evolutionary history? **(2 points)**

<ol>a) Yes</ol>

<ol>b) No</ol>

</ol><br>


<ol>37. Based on the several case studies discussed in lecture, match the convergent mutations for a selected trait in human evolutionary history with the trait they were selected for... **(8 points)**<br>

Traits: lactase persistence, lighter skin pigmentation, hypoxia

<ol>G/C-14010 upstream of *LCT* in Tanzanians</ol>
<ol>Extended haplotype homozygosity around *EGLN1* in Andean populations</ol>
<ol>Missense mutation L-374-F (rs16891982) in the *MATP* gene in Europeans</ol>
<ol>C/G-13907 upstream of *LCT* in Sudanese</ol>
<ol>Missense variant H-615-R (rs1800414) in *OCA2* in East Asians</ol>
<ol>Extended haplotype homozygosity around *EGLN1* in Tibetans</ol>
</ol><br>

<ol>38. Explain, to the best of your ability, why the positive selective sweep around the *PPARA* gene, implicated in high-elevation adaptations in Tibetans, is probably an adaptation related to cold tolerance rather than hypoxia. **(6 points)**</ol><br>

###Part 5: Quantitative Genetics (25 points)

<ol>39. Define *heritability*. **(5 points)**</ol><br>

<ol>40. What is the difference between *broad sense* and *narrow sense* heritability? **(5 points)**</ol><br>

<ol>41. The basic equation for modeling a phenotype lis much like a regression line, and can be described using the equation ***p = Œº + Œ£ Œ≤<sub>i</sub> x<sub>i</sub> + a + e***. If we were to use this equation to estimate the variance components of body mass based on a QTL in *UCP1* and an experimental dietary covariate, what would each component of that equation represent?  Use one term each to match the parameter to what it represents:  **(10 points)**
<br><br>
<ol><ol>Body Mass    |    Diet Covariate    |    Change in Mass due to Alleles at QTL    |    Baseline Body Mass    |    Change in Mass due to Environmental Effects  |  Change in Mass due to Diet </ol></ol>
<br><br>

<ol>p:</ol>

<ol>Œº:</ol>

<ol>Œ≤<sub>i</sub>:</ol>

<ol>x<sub>i</sub>:</ol>

<ol>a:</ol>

<ol>e:</ol>

</ol><br>

<ol>42. If Trait A has a heritability of 1, and Trait B has a heritability of 0.4, which will respond more readily to selection? Using the concept of realized heritability, explain why? **(5 points)**

<center>
##Congratulations!  You've finished your last midterm!
</center>
<br><br><br>
