<!DOCTYPE html>
<html lang="en-US">
<head>
<meta charset="UTF-8" />
<title>Why I&#8217;m wary of candidate gene studies | Slate Lab</title>
<link rel="profile" href="http://gmpg.org/xfn/11" />
<link rel="stylesheet" type="text/css" media="all" href="http://jon-slate.staff.shef.ac.uk/wp-content/themes/twentyten-child-calibri/style.css" />
<link rel="pingback" href="http://jon-slate.staff.shef.ac.uk/xmlrpc.php" />
<link rel='dns-prefetch' href='//fonts.googleapis.com' />
<link rel='dns-prefetch' href='//s.w.org' />
<link rel="alternate" type="application/rss+xml" title="Slate Lab &raquo; Feed" href="http://jon-slate.staff.shef.ac.uk/feed/" />
		<script type="text/javascript">
			window._wpemojiSettings = {"baseUrl":"https:\/\/s.w.org\/images\/core\/emoji\/2\/72x72\/","ext":".png","svgUrl":"https:\/\/s.w.org\/images\/core\/emoji\/2\/svg\/","svgExt":".svg","source":{"concatemoji":"http:\/\/jon-slate.staff.shef.ac.uk\/wp-includes\/js\/wp-emoji-release.min.js?ver=4.6.12"}};
			!function(a,b,c){function d(a){var c,d,e,f,g,h=b.createElement("canvas"),i=h.getContext&&h.getContext("2d"),j=String.fromCharCode;if(!i||!i.fillText)return!1;switch(i.textBaseline="top",i.font="600 32px Arial",a){case"flag":return i.fillText(j(55356,56806,55356,56826),0,0),!(h.toDataURL().length<3e3)&&(i.clearRect(0,0,h.width,h.height),i.fillText(j(55356,57331,65039,8205,55356,57096),0,0),c=h.toDataURL(),i.clearRect(0,0,h.width,h.height),i.fillText(j(55356,57331,55356,57096),0,0),d=h.toDataURL(),c!==d);case"diversity":return i.fillText(j(55356,57221),0,0),e=i.getImageData(16,16,1,1).data,f=e[0]+","+e[1]+","+e[2]+","+e[3],i.fillText(j(55356,57221,55356,57343),0,0),e=i.getImageData(16,16,1,1).data,g=e[0]+","+e[1]+","+e[2]+","+e[3],f!==g;case"simple":return i.fillText(j(55357,56835),0,0),0!==i.getImageData(16,16,1,1).data[0];case"unicode8":return i.fillText(j(55356,57135),0,0),0!==i.getImageData(16,16,1,1).data[0];case"unicode9":return i.fillText(j(55358,56631),0,0),0!==i.getImageData(16,16,1,1).data[0]}return!1}function e(a){var c=b.createElement("script");c.src=a,c.type="text/javascript",b.getElementsByTagName("head")[0].appendChild(c)}var f,g,h,i;for(i=Array("simple","flag","unicode8","diversity","unicode9"),c.supports={everything:!0,everythingExceptFlag:!0},h=0;h<i.length;h++)c.supports[i[h]]=d(i[h]),c.supports.everything=c.supports.everything&&c.supports[i[h]],"flag"!==i[h]&&(c.supports.everythingExceptFlag=c.supports.everythingExceptFlag&&c.supports[i[h]]);c.supports.everythingExceptFlag=c.supports.everythingExceptFlag&&!c.supports.flag,c.DOMReady=!1,c.readyCallback=function(){c.DOMReady=!0},c.supports.everything||(g=function(){c.readyCallback()},b.addEventListener?(b.addEventListener("DOMContentLoaded",g,!1),a.addEventListener("load",g,!1)):(a.attachEvent("onload",g),b.attachEvent("onreadystatechange",function(){"complete"===b.readyState&&c.readyCallback()})),f=c.source||{},f.concatemoji?e(f.concatemoji):f.wpemoji&&f.twemoji&&(e(f.twemoji),e(f.wpemoji)))}(window,document,window._wpemojiSettings);
		</script>
		<style type="text/css">
img.wp-smiley,
img.emoji {
	display: inline !important;
	border: none !important;
	box-shadow: none !important;
	height: 1em !important;
	width: 1em !important;
	margin: 0 .07em !important;
	vertical-align: -0.1em !important;
	background: none !important;
	padding: 0 !important;
}
</style>
<link rel='stylesheet' id='tt-easy-google-fonts-css'  href='http://fonts.googleapis.com/css?family=Raleway%3A100%2Cregular%2C300&#038;subset=latin%2Call&#038;ver=4.6.12' type='text/css' media='all' />
<link rel='stylesheet' id='tablepress-default-css'  href='http://jon-slate.staff.shef.ac.uk/wp-content/plugins/tablepress/css/default.min.css?ver=1.7' type='text/css' media='all' />
<style id='tablepress-default-inline-css' type='text/css'>
.tablepress-id-1 th,.tablepress-id-1 td{padding:10px!important}.tablepress-id-1 .column-1{width:600px}.tablepress-id-1 .column-2{width:200px}.tablepress-id-1 .column-3{width:100px}tablepress-id-2 th,.tablepress-id-2 td{padding:10px!important}.tablepress-id-2 .column-1{width:30px}.tablepress-id-2 .column-2{width:320px}.tablepress-id-2 .column-3{width:100px;width:160px}.tablepress-id-2 .column-2{text-align:justify}
</style>
<link rel='stylesheet' id='cpsh-shortcodes-css'  href='http://jon-slate.staff.shef.ac.uk/wp-content/plugins/column-shortcodes/assets/css/shortcodes.css?ver=0.6.9' type='text/css' media='all' />
<link rel='https://api.w.org/' href='http://jon-slate.staff.shef.ac.uk/wp-json/' />
<link rel="EditURI" type="application/rsd+xml" title="RSD" href="http://jon-slate.staff.shef.ac.uk/xmlrpc.php?rsd" />
<link rel="wlwmanifest" type="application/wlwmanifest+xml" href="http://jon-slate.staff.shef.ac.uk/wp-includes/wlwmanifest.xml" /> 
<link rel='prev' title='In defence of Paula Radcliffe and her biological passport' href='http://jon-slate.staff.shef.ac.uk/in-defence-of-paula-radcliffe-and-her-biological-passport/' />
<link rel='next' title='Comparing PhD Vivas/defences across Europe' href='http://jon-slate.staff.shef.ac.uk/comparing-phd-vivasdefences-across-europe/' />
<meta name="generator" content="WordPress 4.6.12" />
<link rel="canonical" href="http://jon-slate.staff.shef.ac.uk/why-im-wary-of-candidate-gene-studies/" />
<link rel='shortlink' href='http://jon-slate.staff.shef.ac.uk/?p=296' />
<link rel="alternate" type="application/json+oembed" href="http://jon-slate.staff.shef.ac.uk/wp-json/oembed/1.0/embed?url=http%3A%2F%2Fjon-slate.staff.shef.ac.uk%2Fwhy-im-wary-of-candidate-gene-studies%2F" />
<link rel="alternate" type="text/xml+oembed" href="http://jon-slate.staff.shef.ac.uk/wp-json/oembed/1.0/embed?url=http%3A%2F%2Fjon-slate.staff.shef.ac.uk%2Fwhy-im-wary-of-candidate-gene-studies%2F&#038;format=xml" />
<style id="tt-easy-google-font-styles" type="text/css">p { font-family: 'Raleway'; font-style: normal; font-weight: 100; }
h1 { font-family: 'Raleway'; font-style: normal; font-weight: 400; }
h2 { font-family: 'Raleway'; font-style: normal; font-weight: 400; }
h3 { font-family: 'Raleway'; font-style: normal; font-weight: 400; }
h4 { }
h5 { font-family: 'Raleway'; font-style: normal; font-weight: 100; }
h6 { font-family: 'Raleway'; font-style: normal; font-weight: 100; }
body,span,applet,object,iframe,h1,p,blockquote,pre,a,abbr,acronym,address,big,cite,font,h2,h3,h4,h5,h6 { color: #000000; font-family: 'Raleway'; font-style: normal; font-weight: 300; }
</style></head>

<body class="single single-post postid-296 single-format-standard">
<div id="wrapper" class="hfeed">
	<div id="header">
		<div id="masthead">
			<div id="branding" role="banner">
								<div id="site-title">
					<span>
						<a href="http://jon-slate.staff.shef.ac.uk/" title="Slate Lab" rel="home">Slate Lab</a>
					</span>
				</div>
				<div id="site-description">Jon Slate&#039;s Evolutionary Genetics Group</div>

										<img src="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/06/cropped-Newbanner1-e1436472032130.jpg" width="940" height="135" alt="" />
								</div><!-- #branding -->

			<div id="access" role="navigation">
			  				<div class="skip-link screen-reader-text"><a href="#content" title="Skip to content">Skip to content</a></div>
								<div class="menu-header"><ul id="menu-menu-1" class="menu"><li id="menu-item-264" class="menu-item menu-item-type-custom menu-item-object-custom menu-item-home menu-item-264"><a href="http://jon-slate.staff.shef.ac.uk/">Blog</a></li>
<li id="menu-item-265" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-265"><a href="http://jon-slate.staff.shef.ac.uk/about-the-group/">About</a></li>
<li id="menu-item-267" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-267"><a href="http://jon-slate.staff.shef.ac.uk/publications/">Publications</a></li>
<li id="menu-item-268" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-268"><a href="http://jon-slate.staff.shef.ac.uk/slate-group-members/">Past &#038; Present Group Members</a></li>
<li id="menu-item-333" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-333"><a href="http://jon-slate.staff.shef.ac.uk/great-tit-hapmap-project/">Great Tit HapMap Project</a></li>
<li id="menu-item-266" class="menu-item menu-item-type-post_type menu-item-object-page menu-item-266"><a href="http://jon-slate.staff.shef.ac.uk/joining-the-group/">Like to join us?</a></li>
</ul></div>			</div><!-- #access -->
		</div><!-- #masthead -->
	</div><!-- #header -->

	<div id="main">

		<div id="container">
			<div id="content" role="main">

			

				<div id="nav-above" class="navigation">
					<div class="nav-previous"><a href="http://jon-slate.staff.shef.ac.uk/in-defence-of-paula-radcliffe-and-her-biological-passport/" rel="prev"><span class="meta-nav">&larr;</span> In defence of Paula Radcliffe and her biological passport</a></div>
					<div class="nav-next"><a href="http://jon-slate.staff.shef.ac.uk/comparing-phd-vivasdefences-across-europe/" rel="next">Comparing PhD Vivas/defences across Europe <span class="meta-nav">&rarr;</span></a></div>
				</div><!-- #nav-above -->

				<div id="post-296" class="post-296 post type-post status-publish format-standard hentry category-uncategorized">
					<h1 class="entry-title">Why I&#8217;m wary of candidate gene studies</h1>

					<div class="entry-meta">
						<span class="meta-prep meta-prep-author">Posted on</span> <a href="http://jon-slate.staff.shef.ac.uk/why-im-wary-of-candidate-gene-studies/" title="1:22 am" rel="bookmark"><span class="entry-date">November 3, 2015</span></a> <span class="meta-sep">by</span> <span class="author vcard"><a class="url fn n" href="http://jon-slate.staff.shef.ac.uk/author/jon-the-boss/" title="View all posts by Jon Slate">Jon Slate</a></span>					</div><!-- .entry-meta -->

					<div class="entry-content">
						<script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script><p style="text-align: justify;"><strong>Candidate genes</strong></p>
<p style="text-align: justify;">The candidate gene approach to finding genes that determine phenotypic variation is hugely attractive. Rather than performing whole genome scans, involving large amounts of time and money (and a statistical burden of multiple testing), why not just look at those genes which are good candidates? Candidates can be identified by looking through the literature to find genes that influence similar traits in other organisms such as humans, mice, fruit flies or <em>Arabidopsis.</em> There is no doubt that the candidate gene approach has yielded some nice results, perhaps most obviously in studies of plumage and coat colour polymorphism in vertebrates [1-3]. However, at least in the evolutionary genetics literature, most successes are for traits with a simple Mendelian basis, and  where there is a well-established pathway of which genes interact to produce the phenotype of interest. In other words, the candidate gene approach works well for simple traits.</p>
<p style="text-align: justify;">Of course, most traits are complex, and are determined by many genes of small effect. I worry &#8211;  only a little bit, not  in the &#8216;I can&#8217;t sleep at night&#8217; sense &#8211;  that the candidate gene approach applied to these traits can send people on a wild goose chase. As I explain below, I think there are already a number of studies in the molecular ecology field where people are investing too much effort into following up candidate gene results which are, at best, equivocal. The example I&#8217;m going to use comes from studies of the DRD4 gene and it&#8217;s possible role in behavioural variation in wild birds. I&#8217;ve chosen this example, because I think the authors of the original studies have been extremely careful. If the results are not clear cut when this much care is taken, I think the problems I describe will be exacerbated in less rigorous studies (of which there are plenty).</p>
<p style="text-align: justify;"><strong>Population structure &#8211; the known unknown</strong></p>
<p style="text-align: justify;">There are two main statistical problems when assessing whether candidate gene studies are significant. The first is the well known issue that population structure can lead to false positive associations, unless properly controlled for. In a study involving many thousands of tests, it is easy to see when population structure is an issue. A comparison of observed test statistics (or P values) against an expected distribution, shows when datasets are prone to false positives. The slope of the regression of observed against expected values, gives an inflation factor (lambda) which is a measure of the degree of the problem. The plot below shows two lines, each with 100,000 statistical tests &#8211; one where there is no issue (lambda = 1) and another where there is a lot of inflation (lambda = 3), caused by population structure.</p>
<p style="text-align: justify;"><a href="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Q-QPlot.jpeg"><img class="alignnone size-full wp-image-297" src="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Q-QPlot.jpeg" alt="Q-QPlot" width="509" height="358" srcset="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Q-QPlot.jpeg 509w, http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Q-QPlot-300x211.jpeg 300w" sizes="(max-width: 509px) 100vw, 509px" /></a></p>
<p style="text-align: justify;">You might be thinking that an inflation factor of 3 is pretty extreme &#8211; it probably is &#8211; but it certainly isn&#8217;t unprecedented. Susan Johnston (@SuseJohnston) published a very nice paper [4] a few years ago where she performed a GWAS for the age at which wild salmon migrated back to freshwater from the sea. In that study, lambda was 3.24, and without controlling for the genetic structure there would have been many false positive results. We can see the effect of failing to control for structure below:</p>
<p style="text-align: justify;">Here, I have simulated 100,000 datapoints (imaginary SNPs), assuming either no structure (lambda = 1) or salmon-like structure (lambda = 3). The plot below shows the distribution of chi square values following association testing. For clarity, I&#8217;ve restricted the x-axis to range from 0.5 to 20, but in fact the difference between the histograms is more extreme than it looks, because the structured dataset has a long tail, while more than half of the lambda = 1 datapoints have chisquare values &lt; 0.5. In the structured dataset, if we (naiively) assumed our test statistic followed a chi square distribution with 1 df we would have a massive false positive problem. Almost 26% of the datapoints are &#8216;significant&#8217; at P &lt; 0.05, 13.7% are significant at P &lt; 0.01 and 5.7% are significant at P &lt; 0.001. In other words, we are in real danger of assigning biological importance to what are actually chance results.</p>
<p style="text-align: justify;"><a href="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/ChiSquareDistributions.jpeg"><img class="alignnone size-full wp-image-298" src="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/ChiSquareDistributions.jpeg" alt="ChiSquareDistributions" width="509" height="358" srcset="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/ChiSquareDistributions.jpeg 509w, http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/ChiSquareDistributions-300x211.jpeg 300w" sizes="(max-width: 509px) 100vw, 509px" /></a></p>
<p style="text-align: justify;">Fortunately, when we perform GWAS studies, there are methods for controlling for the population genetic structure &#8211; the simplest is just dividing the test statistic by the inflation factor before calculating the P value. There are other more sophisticated approaches, but they all do a good job of dealing with the effects of structure. So what&#8217;s the problem for candidate gene studies then?</p>
<p style="text-align: justify;">The problem is that we only know we have a problem of inflated test statistics when we have performed genomewide scans with many independent datapoints. When we do a candidate gene study, we typically look at one or a few genes, and so it is virtually impossible to know whether our significant results are real or simply an artefact of unappreciated genetic structure causing false positives.</p>
<p style="text-align: justify;">So, one of the problems with candidate gene studies is that too few statistical tests have been performed for us to be able to say whether a gene really does have an effect on the trait. Strangely, the second statistical problem can arise when we have performed relatively few, yet still too many, tests and the multiple testing problem can make it difficult to know whether a result is statistically significant or not. Typically, this happens when multiple SNPs are examined, sometimes in tandem with several traits and several different genetic models. The DRD4 studies are informative in this regard.</p>
<p style="text-align: justify;"><strong>DRD4 in wild birds &#8211; background</strong><span style="text-decoration: underline;"><strong><br />
</strong></span></p>
<p style="text-align: justify;">The paper that started the enthusiasm for studying DRD4 in birds is by Fidler and colleagues and was published in 2007 in Proc. R. Soc. B. [5]. The paper has been very influential; at the time of writing it has been cited &gt; 80 times. Briefly, the authors studied novelty-seeking behaviour in great tits, which was recorded by observing how long it took birds to explore novel objects in a cage or room. Data were collected on hand-reared wild birds, and on birds from selection lines for fast or slow &#8216;early exploratory behaviour&#8217; (EEB). EEB had previously been shown to be heritable, and to respond to artifical selection in this population, hence the existence of divergent lines. The authors reasoned that the dopamine receptor D4 gene (DRD4) was a good candidate for explaining variation in behaviour because it has been associated with novelty-seeking behaviour in humans (see refs within Fidler).</p>
<p style="text-align: justify;">The authors sequenced the full coding sequence of DRD4 in great tits &#8211; no easy task. In total they found 73 polymorphsism (66 SNPs and 7 indels). However, most of their attention was on the 3rd exon, because it was this region that was associated with relevant traits in other vertebrates. They first showed that the exon contained a single synonymous SNP (hereafter SNP830) which differed in frequency between the fast (n=29) and slow (n=21) lines. As the authors pointed out, genetic drift could cause these kinds of effects. Therefore, they analysed a further 91 hand-reared but unselected birds that were wild-caught as nestlings, from 17 nests.</p>
<p style="text-align: justify;">The results of an association study on the wild birds are shown below &#8211; this figure is taken directly from the paper. Genotype TT is associated with the highest EEB score, which is consistent with the trend from the selection lines. This study was conducted in the days before many of the ways we correct for population structure were fully developed, but the authors were well aware of the problem. It is unlikely that there was much structure in the data because great tits have a very large effective population size, there is very low genetic structure across Europe, and the birds were sampled from a single location. However, by sampling 91 birds from just 17 nests, structure is introduced by the inclusion of relatives. The authors recognised this problem, by fitting nest as a random effect in their models. Today, we may take a slightly different approach by fitting a relationship matrix (derived from the pedigree or markers) in the model, but the approach Fidler and colleagues took here probably deals with the problem ok, and it certainly demonstrates an awareness of the problem and how to deal with it.</p>
<p style="text-align: justify;"><a href="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/FidlerFig2.png"><img class="alignnone wp-image-299 size-full" src="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/FidlerFig2.png" alt="FidlerFig2" width="493" height="678" srcset="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/FidlerFig2.png 493w, http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/FidlerFig2-218x300.png 218w" sizes="(max-width: 493px) 100vw, 493px" /></a></p>
<p style="text-align: justify;">At face value, this looks like a pretty good story. The selection line and wild bird analyses are consistent, and the SNP is in the functionally relevant part of the gene. It&#8217;s worth stating here though, that the authors actually fitted three different genetic models, and they also typed an indel in the wild birds, so they actually performed 6 different (but perhaps not independent) tests on the data, which does make the statistical significance pretty marginal (even ignoring the possibility of inflated lambda). To their credit, Fidler and colleagues raised the possibility that the results could be an artefact, caused by population genetic structure. I have no beef with this paper &#8211; it was ground-breaking, potentially important, and yet circumspect. However, my worry is that subsequent studies  seem to make an assumption that the DRD4 story is both robust and repeatable elsewhere. I don&#8217;t think it is.</p>
<p style="text-align: justify;"><strong>DRD4 in great tits &#8211; followup studies</strong></p>
<p style="text-align: justify;">The first follow-up paper to Fidler was by the same group &#8211; this time led by Peter Korsten [6]. This is a really nice example of an attempt to replicate the original results. Not only do the authors type a new set of wild birds from the original population (Westerheide, in the Netherlands), but they also look at three other great tit populations &#8211; Wytham Woods in Oxford, Boshoek in Belgium and Lauwersmeer in the Netherlands. Similar tests of exploratory behaviour were examined in each population. The orginal SNP &#8211; SNP830 &#8211; and the indel that significantly differed between the fast and slow EEB lines in Fidler were typed in all 4 populations.</p>
<p style="text-align: justify;">The results from Korsten et al are quite striking. First of all, the SNP830 result in Westerheide was replicated in the new set of 77 birds sampled from the wild. Second, the associations could not be replicated in any of the other 3 populations. An analysis pooling data from all 4 populations did not show an association between SNP830 and exploratory behaviour. There was a significant interaction term between SNP genotype and population, but this could mean either there are genuine gene by environment interactions, or the original Westerheide result was a false positive and the others are non-significant. Again, the association between Westerheide and SNP830 is only nominally significant, and if multiple tests for the number of loci, populations or genetic models were performed, it would be non-significant. Regardless though, the evidence for SNP830 being associated with EEB in this population is strengthened by this study.</p>
<p style="text-align: justify;">One of the puzzles from Korsten et al is why wasn&#8217;t the original result replicated elsewhere? One possiblilty that the authors suggested is that SNP830 isn&#8217;t the causal mutation, but is in LD with an unknown causal mutation closeby. If the LD structure varied between populations, then one might not expect to see similar associations with SNP830 in other populations. Another possibility is that there is no real association in any of the populations, including Westerheide.</p>
<p style="text-align: justify;">In 2013, the same authors (led by Jakob Mueller) typed a much larger number of polymorphisms in the DRD4 region, in the same birds as used by Korsten [7]. Again, this is a very substantial and impressive piece of work. This time, around 98 polymorphisms were studied, which made it possible to define linkage disequilibrium around DRD4 and to make comparisons in pattern of LD between populations. In addition, there were now a large enough number of tests to get some feel for whether test statistics were inflated due to e.g. the population structure or the presence of relatives. In all 4 populations, LD declined quite rapidly, but patterns were quite similar across populations. This means that (i) the different SNPs can be regarded as fairly independent tests (especially if they are in different haploblocks) and (ii) the suggestion that Westerheide shows an assocation with SNP830, but the others don&#8217;t, due to heterogeneity in the LD between SNP830 and a causal variant, is not really supported.</p>
<p style="text-align: justify;">The data from Mueller et al are on Dryad, and so I have had a look at them in a little more detail. I was interested to know whether (i) there is much evidence for test statistic inflation (i.e. lambda &gt;1), which could be driving false positive results and (ii) whether the significant results look very compelling. For simplicity, I have only looked at additive models for a couple of reasons. First, the original SNP830 association in Westerheide was more significant using this model compared to models that assumed dominance or overdominance. Second, I don&#8217;t find the reasons for fitting overdominance models very compelling, although the reasons for that can wait for a future post.</p>
<p style="text-align: justify;">I followed the model fitting procedure used by Mueller; SNP genotypes were coded 0,1 or 2 and fitted as a covariate in models of association with personality score. For the Westerheide population brood was fitted as a random effect. Plots showing observed against expected p values (assuming a uniform distriubtion of P values between 0 and 1) are shown below. Each SNP is a separate datapoint and SNP830 is shown as a red dot.</p>
<p style="text-align: justify;"><a href="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Westerheide_All.jpeg"><img class="alignnone wp-image-304 size-medium" src="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Westerheide_All-300x211.jpeg" alt="Westerheide_All" width="300" height="211" srcset="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Westerheide_All-300x211.jpeg 300w, http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Westerheide_All.jpeg 509w" sizes="(max-width: 300px) 100vw, 300px" /></a></p>
<p style="text-align: justify;"><a href="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Westerheide2007.jpeg"><img class="alignnone wp-image-300 size-medium" src="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Westerheide2007-300x211.jpeg" alt="Westerheide2007" width="300" height="211" srcset="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Westerheide2007-300x211.jpeg 300w, http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Westerheide2007.jpeg 509w" sizes="(max-width: 300px) 100vw, 300px" /></a><a href="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Boshoek.jpeg"><img class="alignnone wp-image-301 size-medium" src="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Boshoek-300x211.jpeg" alt="Boshoek" width="300" height="211" srcset="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Boshoek-300x211.jpeg 300w, http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Boshoek.jpeg 509w" sizes="(max-width: 300px) 100vw, 300px" /></a></p>
<p style="text-align: justify;"><a href="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Wytham.jpeg"><img class="alignnone wp-image-302 size-medium" src="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Wytham-300x211.jpeg" alt="Wytham" width="300" height="211" srcset="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Wytham-300x211.jpeg 300w, http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Wytham.jpeg 509w" sizes="(max-width: 300px) 100vw, 300px" /></a><a href="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Lauwersmeer.jpeg"><img class="alignnone wp-image-303 size-medium" src="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Lauwersmeer-300x211.jpeg" alt="Lauwersmeer" width="300" height="211" srcset="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Lauwersmeer-300x211.jpeg 300w, http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2015/11/Lauwersmeer.jpeg 509w" sizes="(max-width: 300px) 100vw, 300px" /></a></p>
<p style="text-align: justify;">Let&#8217;s first look at the Westerheide population &#8211; the combined data from both years (i.e. the Fidler study and the Korsten study) shows that SNP830 was more strongly associated with exploratory behaviour than any other locus. Not only that, but the data fit pretty well on the lambda = 1 line, but with SNP830 some way above it. This looks pretty convincing to me, although compared to the most significant loci in a genome wide association study, it is not very significant. It&#8217;s also noteworthy that when the data are restricted to birds sampled in 2007 &#8211; these are the birds in the Korsten follow-up study &#8211; SNP830 is still the most significantly associated SNP.</p>
<p style="text-align: justify;">In the three other populations, SNP830 is unremarkable. There is no evidence whatsoever that it is associated with exploratory behaviour. The other populations do not show serious problems with inflation of test statistics, although there is a hint of it in the Wytham Woods population. The most significant SNP in Lauwersmeer is more significant than expected by chance, but not by much. It should be said, that none of these SNPs reach significance after applying a Bonferrroni correction or similar procedure.</p>
<p style="text-align: justify;">On balance then, I think the SNP830 association in Westerheide looks reasonably convincing, but I don&#8217;t think associations between DRD4 and exploratory behaviour in other great tit populations have been demonstrated. The authors of these papers have always been very careful to highlight alternative explanations, and overall it&#8217;s an impressive body of work. What I find surprising, is that despite these fairly equivocal results, there seems to be a slew of follow-up DRD4 candidate gene studies in other bird species, with what look to me, to be fairly wishful interpretations of the results. I worry that a narrative is building up that DRD4 explains lots of variation in personality traits in lots of different bird species. What do these other papers show?</p>
<p style="text-align: justify;"><strong>DRD4 in other bird populations</strong></p>
<p style="text-align: justify;">I&#8217;ll briefly discuss some of the other studies.</p>
<p style="text-align: justify;">Gillingham and colleagues looked at DRD4 in relation to body condition in flamingoes [8]. They argue that DRD4 genotypes are associated with the phenotype, but it is pretty hard to tell whether there is much going on. They use an unconventional (for association studies) approach of AIC-based multi-model parameter estimation and they only test DRD4. I dislike this approach for association studies, because by studying a single locus (as is the case here) or fitting one locus at a time it is really easy for a locus that doesn&#8217;t affect a trait to appear to explain a few % of the variation. The authors actually typed 10 microsatellites, and it is my guess that fitting genotypes at the microsatellites would have yielded similar results to DRD4. Certainly, I think the claim that <em>&#8220;This is to our knowledge, the first study to show an association between exon 3 DRD4 polymorphism and body condition in non-human animals&#8221; </em>needs some more validation before it can be believed.</p>
<p style="text-align: justify;">In a study of adaptation to urban environments in blackbirds [9], Mueller et al (2013) examined 16 candidate polymorphisms (including 5 in DRD4) in 12 populations, but there was no compelling evidence that DRD4 played a role in adaptation.</p>
<p style="text-align: justify;">Mueller and colleagues also scored the effects of DRD4 polymorphisms in exploratory behaviour in yellow-crowned bishops [10]. They replicated the study in Portugese and Spanish populations. The sample sizes were fairly small, nor was there any control for population structure. A large proportion of SNPs were non-significant, but two SNPs were associated with behavioural variation. One of these SNPs was significant in both populations; the other was significant in one population and approached significance in the other. The directions of the associations were the same in both populations If both SNPs had been significant in both populations (they narrowly missed), the result would have been experiment-wide significant. This study looks worthy of follow-up, although I&#8217;d be a little wary of population structure being an issue given the demographic history of this species and the way the birds were sampled/obtained. The jury&#8217;s still out for me.</p>
<p style="text-align: justify;">Garamszegi and colleagues looked at two DRD4 SNPs in relation to three behavioural traits in collared flycatchers [11]. One association was significant and another marginally so &#8211; again though, we know very little about what test statistics would be generated by other loci and whether population structure is a confounding issue.</p>
<p style="text-align: justify;">In a study of invasive starling populations in Australia [12], Rollins used propensity to enter a trap as a proxy for boldness. There were no significant associations between DRD4 and boldness &#8211; sample sizes were quite large in this study.</p>
<p style="text-align: justify;">There has been a new great tit study this year [13], with an Estonian population typed at SNP830 and measured for feeding times/delays when presented with a novel object near the nest. In males, birds with the CC genotype delayed feeding longer than other birds, but the effect was not observed in females. The sample sizes were modest. It&#8217;s a tantalising result.</p>
<p style="text-align: justify;">Finally, an attempt to examine DRD4 in relation to personality traits in Seychelles Warblers found no DRD4 polymorphisms [14].</p>
<p style="text-align: justify;"><strong>Final Thoughts</strong></p>
<p style="text-align: justify;">Overall, I think some of the follow up studies provide <em>some, tentative</em> evidence for associations between DRD4 and personality traits, but I&#8217;m not convinced that any randonmly chosen gene would have given much weaker evidence.  Would I take a candidate gene approach to studying behavioural variation in birds? Probably not. If we cannot replicate results across different great tit populations (where genetic structure is low), it doesn&#8217;t seem likely that we can easily detect variation in other species. Is there good evidence that DRD4 explains behavioural variation in other bird species? I don&#8217;t think so.</p>
<p style="text-align: justify;">I do think that the main problem with candidate gene studies is the more general one, that we just don&#8217;t know <em>how</em> significant something is until we have seen data from across the genome. Fortunately, with whole genome scans becoming ever easier, the need to even perform candidate gene studies is diminishing. For what its worth, I would avoid them unless the trait has a really simple Mendelian basis and the genetic pathways underlying the trait are well understood.</p>
<h4 style="text-align: justify;"><strong>References</strong></h4>
<h5 style="text-align: justify;"><strong>1.</strong> Nachman et al. (2003) The genetic basis of adaptive melanism in pocket mice. <span style="text-decoration: underline;">PNAS</span> 100: <span class="btext">5268–5273</span><br />
<strong>2.</strong> Mundy et al. (2004) Conserved genetic basis of a quantitative plumage trait involved in mate choice. <span style="text-decoration: underline;">Science</span> 303: 1870-1873<br />
<strong>3.</strong> Gratten et al (2007) Compelling evidence that a single nucleotide polymorphism in TYRP1 is responsible for coat colour polymorphism in a free-living population of Soay sheep. <span style="text-decoration: underline;">Proc. R. Soc. Lond B</span>. 274: 619-626<br />
<strong>4.</strong> Johnston et al. (2014). Genome-wide SNP analysis reveals a genetic basis for sea-age variation in a wild population of Atlantic salmon (<em>Salmo salar</em>). <span style="text-decoration: underline;">Molecular Ecology</span> 23:3452-3468<br />
<strong>5.</strong> Fidler et al. (2007) Drd4 gene polymorphisms are associated with personality variation in a passerine bird. <span style="text-decoration: underline;">Proc. R. Soc. Lond B</span>. 274: 1685-1691.<br />
<strong>6.</strong> Korsten et al. (2010) Association between DRD4 gene polymorphism and personality variation in great tits: a test across four wild populations. <span style="text-decoration: underline;">Molecular Ecology</span> 19: 832-843.<br />
<strong>7.</strong> Mueller et al (2013) Haplotype structure, adaptive history and associations with exploratory behaviour of the DRD4 gene region in four great tit (<em>Parus major</em>) populations. <span style="text-decoration: underline;">Molecular Ecology</span> 22:2797-2808<br />
<strong>8.</strong> Gillingham et al (2012) Genetic polymorphism in dopamine receptor D4 is associated with early body condition in a large population of greater flamingos, <em>Phoenicopterus roseus.</em> <span style="text-decoration: underline;">Molecular Ecology</span> 21:4024-4037.<br />
<strong>9.</strong> Mueller et al (2013) Candidate gene polymorphisms for behavioural adaptations during urbanization in blackbirds. <span style="text-decoration: underline;">Molecular Ecology</span> 22:3629-3637<br />
<strong>10.</strong> Mueller et al (2014) Behaviour-related DRD4 polymorphisms in invasive bird populations. <span style="text-decoration: underline;">Molecular Ecology</span> 23:2876-2885<br />
<strong>11.</strong> Garamszegi et al (2014) The relationship between DRD4 polymorphisms and phenotypic correlations of behaviors in the collared flycatcher. <span style="text-decoration: underline;">Ecology and Evolution</span> 4: 1466–1479<br />
<strong>12.</strong> Rollins et al (2015) Is there evidence of selection in the dopamine receptor D4 gene in Australian invasive starling populations? <span style="text-decoration: underline;">Current Zoology</span> 61: 505–519<br />
<strong>13.</strong> Timm et al (2015) DRD4 gene polymorphism in great tits: gender-specific association with behavioural variation in the wild. <span style="text-decoration: underline;">Behavioural Ecology and Sociobiology</span> 69:729–735<br />
<strong>14.</strong> Edwards et al (2015) No Association between Personality and Candidate Gene Polymorphisms in a Wild Bird Population. <span style="text-decoration: underline;">PloS One</span> 10: e0138439.</h5>
											</div><!-- .entry-content -->


					<div class="entry-utility">
						This entry was posted in <a href="http://jon-slate.staff.shef.ac.uk/category/uncategorized/" rel="category tag">Uncategorized</a>. Bookmark the <a href="http://jon-slate.staff.shef.ac.uk/why-im-wary-of-candidate-gene-studies/" title="Permalink to Why I&#8217;m wary of candidate gene studies" rel="bookmark">permalink</a>.											</div><!-- .entry-utility -->
				</div><!-- #post-## -->

				<div id="nav-below" class="navigation">
					<div class="nav-previous"><a href="http://jon-slate.staff.shef.ac.uk/in-defence-of-paula-radcliffe-and-her-biological-passport/" rel="prev"><span class="meta-nav">&larr;</span> In defence of Paula Radcliffe and her biological passport</a></div>
					<div class="nav-next"><a href="http://jon-slate.staff.shef.ac.uk/comparing-phd-vivasdefences-across-europe/" rel="next">Comparing PhD Vivas/defences across Europe <span class="meta-nav">&rarr;</span></a></div>
				</div><!-- #nav-below -->

				

			</div><!-- #content -->
		</div><!-- #container -->


		<div id="primary" class="widget-area" role="complementary">
			<ul class="xoxo">

		<li id="recent-posts-2" class="widget-container widget_recent_entries">		<h3 class="widget-title">Recent Posts</h3>		<ul>
					<li>
				<a href="http://jon-slate.staff.shef.ac.uk/gordon-strachan-genetics-of-height-and-fitba/">Gordon Strachan, genetics of height and fitba</a>
						</li>
					<li>
				<a href="http://jon-slate.staff.shef.ac.uk/a-moan-about-executive-search-firms-and-academic-posts/">A moan about &#8216;Executive Search Firms&#8217; and academic posts</a>
						</li>
					<li>
				<a href="http://jon-slate.staff.shef.ac.uk/zebra-finch-supergene-and-sperm/">Zebra finch supergene and sperm</a>
						</li>
					<li>
				<a href="http://jon-slate.staff.shef.ac.uk/comparing-phd-vivasdefences-across-europe/">Comparing PhD Vivas/defences across Europe</a>
						</li>
					<li>
				<a href="http://jon-slate.staff.shef.ac.uk/why-im-wary-of-candidate-gene-studies/">Why I&#8217;m wary of candidate gene studies</a>
						</li>
				</ul>
		</li>		<li id="archives-2" class="widget-container widget_archive"><h3 class="widget-title">Archives</h3>		<ul>
			<li><a href='http://jon-slate.staff.shef.ac.uk/2017/10/'>October 2017</a></li>
	<li><a href='http://jon-slate.staff.shef.ac.uk/2017/09/'>September 2017</a></li>
	<li><a href='http://jon-slate.staff.shef.ac.uk/2017/07/'>July 2017</a></li>
	<li><a href='http://jon-slate.staff.shef.ac.uk/2016/04/'>April 2016</a></li>
	<li><a href='http://jon-slate.staff.shef.ac.uk/2015/11/'>November 2015</a></li>
	<li><a href='http://jon-slate.staff.shef.ac.uk/2015/09/'>September 2015</a></li>
	<li><a href='http://jon-slate.staff.shef.ac.uk/2015/07/'>July 2015</a></li>
		</ul>
		</li><li id="pages-2" class="widget-container widget_pages"><h3 class="widget-title">Pages</h3>		<ul>
			<li class="page_item page-item-217"><a href="http://jon-slate.staff.shef.ac.uk/about-the-group/">About</a></li>
<li class="page_item page-item-324"><a href="http://jon-slate.staff.shef.ac.uk/great-tit-hapmap-project/">Great Tit HapMap Project</a></li>
<li class="page_item page-item-253"><a href="http://jon-slate.staff.shef.ac.uk/joining-the-group/">Like to join us?</a></li>
<li class="page_item page-item-196"><a href="http://jon-slate.staff.shef.ac.uk/slate-group-members/">Past &#038; Present Group Members</a></li>
<li class="page_item page-item-116"><a href="http://jon-slate.staff.shef.ac.uk/publications/">Publications</a></li>
		</ul>
		</li><li id="text-5" class="widget-container widget_text"><h3 class="widget-title">ResearcherID</h3>			<div class="textwidget">
<span id='badgeCont872772' style='width:126px'><script src='http://labs.researcherid.com/mashlets?el=badgeCont872772&mashlet=badge&showTitle=false&className=a&rid=D-2925-2012'></script></span></div>
		</li><li id="widget_sp_image-2" class="widget-container widget_sp_image"><h3 class="widget-title">GoogleScholar</h3><a href="https://scholar.google.co.uk/citations?user=GpPHIZ8AAAAJ&#038;hl=en" id="" target="_self" class="widget_sp_image-image-link" title="GoogleScholar" rel=""><img width="120" height="120" alt="GoogleScholar" class="attachment-full" style="max-width: 100%;" src="http://jon-slate.staff.shef.ac.uk/wp-content/uploads/2016/09/Google-Scholar-logo-e1474465922541.png" /></a><div class="widget_sp_image-description" ><p><a href="https://scholar.google.co.uk/citations?user=GpPHIZ8AAAAJ&hl=en"></a></p>
</div></li><li id="text-2" class="widget-container widget_text">			<div class="textwidget">            <a class="twitter-timeline"  href="https://twitter.com/Jon_Slate" data-widget-id="616374614944579585">Tweets by @Jon_Slate</a>
            <script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+"://platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>
          </div>
		</li><li id="text-4" class="widget-container widget_text"><h3 class="widget-title">Strava</h3>			<div class="textwidget"><iframe height='480' width='220' frameborder='0' allowtransparency='true' scrolling='no' src='https://www.strava.com/athletes/1675303/latest-rides/2ddd814aa87a5b02e43e0e8329cab2e380a53b16'></iframe></div>
		</li><li id="meta-2" class="widget-container widget_meta"><h3 class="widget-title">Meta</h3>			<ul>
						<li><a href="http://jon-slate.staff.shef.ac.uk/wp-login.php">Log in</a></li>
			<li><a href="http://jon-slate.staff.shef.ac.uk/feed/">Entries <abbr title="Really Simple Syndication">RSS</abbr></a></li>
			<li><a href="http://jon-slate.staff.shef.ac.uk/comments/feed/">Comments <abbr title="Really Simple Syndication">RSS</abbr></a></li>
			<li><a href="https://wordpress.org/" title="Powered by WordPress, state-of-the-art semantic personal publishing platform.">WordPress.org</a></li>			</ul>
			</li>			</ul>
		</div><!-- #primary .widget-area -->

	</div><!-- #main -->

	<div id="footer" role="contentinfo">
		<div id="colophon">



			<div id="site-info">
				<a href="http://jon-slate.staff.shef.ac.uk/" title="Slate Lab" rel="home">
					Slate Lab				</a>
			</div><!-- #site-info -->

			<div id="site-generator">
								<a href="https://wordpress.org/" title="Semantic Personal Publishing Platform">Proudly powered by WordPress.</a>
			</div><!-- #site-generator -->

		</div><!-- #colophon -->
	</div><!-- #footer -->

</div><!-- #wrapper -->

<script type='text/javascript' src='http://jon-slate.staff.shef.ac.uk/wp-includes/js/wp-embed.min.js?ver=4.6.12'></script>
</body>
</html>
